opm/simulators/wells/WellGroupHelpers.cpp

```diff
@@ -97,6 +98,13 @@ namespace Opm {
             const auto& groupTmp = schedule.getGroup(groupName, reportStepIdx);
             const auto& gefac = groupTmp.getGroupEfficiencyFactor();
             rate += gefac * sumWellPhaseRates(res_rates, groupTmp, schedule, wellState, reportStepIdx, phasePos, injector);
+            // only sum satelite production once
+            if (wellState.isRank0() && !injector) {
+                const auto rateComp = selectRateComponent(wellState.phaseUsage(), phasePos);
+                if (rateComp.has_value()) {
+                    rate += satelliteProduction(schedule[reportStepIdx], group.groups(), *rateComp);
+                }
+            }
```

Comment:
This section needs to be outside of the `groupName` loop lest we include the contributions multiple times in the case that more than one of `group.groups()` is a satellite.